HLTrigger/JetMET/plugins/HLTJetHFCleaner.h

```diff
@@ -0,0 +1,213 @@
+  const edm::EDGetTokenT<edm::ValueMap<int>> centralEtaStripSizeToken_;
+
+  // Minimum jet pt to consider while applying HF cuts
+  const float jetPtMin_;
+  const float dphiJetMetMin_;
+
+  // Cuts will be applied to jets only with jetEtaMin_ < |eta| < jetEtaMax_
+  const float jetEtaMin_;
+  const float jetEtaMax_;
+
+  const float sigmaEtaPhiDiffMax_;
+  const float centralEtaStripSizeMax_;
+  const float cornerCutSigmaEtaEta_;
+  const float cornerCutSigmaPhiPhi_;
+  const bool applySigmaEtaPhiCornerCut_;
+  const bool applySigmaEtaPhiCut_;
+  const bool applyStripSizeCut_;
+};
+
+template <typename JetType>
+HLTJetHFCleaner<JetType>::HLTJetHFCleaner(const edm::ParameterSet& iConfig)
+    : jetToken_(consumes(iConfig.getParameter<edm::InputTag>("jets"))),
+      metToken_(consumes(iConfig.getParameter<edm::InputTag>("mets"))),
+      sigmaEtaEtaToken_(consumes(iConfig.getParameter<edm::InputTag>("sigmaEtaEta"))),
+      sigmaPhiPhiToken_(consumes(iConfig.getParameter<edm::InputTag>("sigmaPhiPhi"))),
+      centralEtaStripSizeToken_(consumes(iConfig.getParameter<edm::InputTag>("centralEtaStripSize"))),
+      jetPtMin_(iConfig.getParameter<double>("jetPtMin")),
+      dphiJetMetMin_(iConfig.getParameter<double>("dphiJetMetMin")),
+      jetEtaMin_(iConfig.getParameter<double>("jetEtaMin")),
+      jetEtaMax_(iConfig.getParameter<double>("jetEtaMax")),
+      sigmaEtaPhiDiffMax_(iConfig.getParameter<double>("sigmaEtaPhiDiffMax")),
+      centralEtaStripSizeMax_(iConfig.getParameter<int>("centralEtaStripSizeMax")),
+      cornerCutSigmaEtaEta_(iConfig.getParameter<double>("cornerCutSigmaEtaEta")),
+      cornerCutSigmaPhiPhi_(iConfig.getParameter<double>("cornerCutSigmaPhiPhi")),
+      applySigmaEtaPhiCornerCut_(iConfig.getParameter<bool>("applySigmaEtaPhiCornerCut")),
+      applySigmaEtaPhiCut_(iConfig.getParameter<bool>("applySigmaEtaPhiCut")),
+      applyStripSizeCut_(iConfig.getParameter<bool>("applyStripSizeCut")) {
+  produces<JetCollection>();
+}
+
+template <typename JetType>
+void HLTJetHFCleaner<JetType>::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup&) const {
+  auto const jets = iEvent.getHandle(jetToken_);
+  auto const& mets = iEvent.get(metToken_);
+
+  // The set of cleaned jets
+  auto cleaned_jets = std::make_unique<JetCollection>();
+  cleaned_jets->reserve(jets->size());
+
+  // Check if length of the MET vector is 1, throw exception otherwise
+  if (mets.size() != 1) {
+    throw cms::Exception("ComparisonFailure") << "Size of reco::MET collection is different from 1";
+  }
+  auto const met_phi = mets.front().phi();
+
+  // Loop over the jets and do cleaning
+  for (uint ijet = 0; ijet < jets->size(); ijet++) {
+    auto const jet = (*jets)[ijet];
+    auto const withinEtaRange = (std::abs(jet.eta()) < jetEtaMax_) and (std::abs(jet.eta()) > jetEtaMin_);
+    // Cuts are applied to jets that are back to back with MET
+    if (reco::deltaPhi(jet.phi(), met_phi) <= dphiJetMetMin_) {
```

Comment:
Should this be `abs(deltaPhi)` ?

Reply:
Thanks a lot for catching this, it should indeed be the `abs` value. I fixed this in [ff94a85](https://github.com/cms-sw/cmssw/pull/37539/commits/ff94a85983bfb7cbbb37f7dc76656d354bff00d8)